conftest.py

```diff
@@ -27,3 +28,95 @@ def pytest_addoption(parser):
 def pytest_runtest_setup(item):
     if item.get_closest_marker("itunes") and not item.config.getoption("--itunes"):
         pytest.skip("Test requires --itunes")
+
+
+# XXX: The below code is vendored code from https://github.com/utgwkk/pytest-github-actions-annotate-failures
+# so that we can add support for pytest_rerunfailures
+# retried tests will no longer be annotated in GHA
+#
+# Reference:
+# https://docs.pytest.org/en/latest/writing_plugins.html#hookwrapper-executing-around-other-hooks
+# https://docs.pytest.org/en/latest/writing_plugins.html#hook-function-ordering-call-example
+# https://docs.pytest.org/en/stable/reference.html#pytest.hookspec.pytest_runtest_makereport
+#
+# Inspired by:
+# https://github.com/pytest-dev/pytest/blob/master/src/_pytest/terminal.py
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    report = outcome.get_result()
+
+    # enable only in a workflow of GitHub Actions
+    # ref: https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables
+    if os.environ.get("GITHUB_ACTIONS") != "true":
+        return
+
+    try:
+        # If we have the pytest_rerunfailures plugin,
+        # and there are still retries to be run,
+        # then do not return the error
+        import pytest_rerunfailures
+
+        if item.execution_count <= pytest_rerunfailures.get_reruns_count(item):
+            return
+    except ImportError:
+        pass
```

Comment:
This is the only thing that's changed from the original source: https://github.com/utgwkk/pytest-github-actions-annotate-failures/blob/master/pytest_github_actions_annotate_failures/plugin.py